### PR TITLE
[video_player] BehindLiveWindowException

### PR DESCRIPTION
**Pull request by Ezaldeen99**
*Behind window exceptions that happens in live streaming are really annoying and it maybe a bad UX that the user will get an error each time it happens.*

Fixes 
https://github.com/flutter/flutter/issues/100478.
https://github.com/flutter/flutter/issues/81815

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/video_player/video_player_android/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.3.2
+
+* Handle `BehindLiveWindowException`
```
Comment (stuartmorgan) [inline packages/video_player/video_player_android/CHANGELOG.md:3]: Please fix this to follow the CHANGELOG style guide that was linked in the PR checklist. You've checked the box, but aren't actually following the style described there.

**Comment by stuartmorgan** [comment]
I've updated the checklist to match the actual state of the PR; please re-check the boxes once you've done those steps.

**Comment by Ezaldeen99** [comment]
Thanks for your review, will do the requested changes

File: packages/video_player/video_player_android/lib/src/android_video_player.dart [hunk 2]
```diff
@@ -143,6 +143,7 @@ class AndroidVideoPlayer extends VideoPlayerPlatform {
             buffered: values.map<DurationRange>(_toDurationRange).toList(),
             eventType: VideoEventType.bufferingUpdate,
           );
+        case 'behindLiveWindow':
```
Comment (stuartmorgan) [inline packages/video_player/video_player_android/lib/src/android_video_player.dart:146]: Why is this new event necessary? Is ExoPlayer really starting buffering without calling the listener we have registered?

File: packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java [hunk 3]
```diff
@@ -209,7 +210,18 @@ public void onPlaybackStateChanged(final int playbackState) {
           @Override
           public void onPlayerError(final ExoPlaybackException error) {
             setBuffering(false);
-            if (eventSink != null) {
+            if (isBehindLiveWindow(error) && exoPlayer != null) {
```
Comment (stuartmorgan) [inline packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java:213]: All of the changes in this file--which is the core of the PR--could be reverted without any tests failing, so the PR doesn't have meaningful test coverage. This needs a native unit test.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
@Ezaldeen99 Are you still planning to update this PR to address the latest review comments?

**Comment by Ezaldeen99** [comment]
> @Ezaldeen99 Are you still planning to update this PR to address the latest review comments?

sorry for latency, I will try to update this PR by the end of the week, thanks for understanding

**Comment by stuartmorgan** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR if you have the time to address the review comments. Thanks!

**Pull request closed without merging**